src/app/components/shared/formly/license-input.component.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  OnInit,
+  signal,
+  ViewChild,
+} from "@angular/core";
+import { FieldType } from "@ngx-formly/core";
+import { NgbModal } from "@ng-bootstrap/ng-bootstrap";
+import {
+  LicensesService,
+  SpdxLicense,
+} from "@services/licenses/licenses.service";
+import { asFormControl } from "./helper";
+
+@Component({
+  selector: "baw-license-input",
+  template: `
+    <div class="form-group mb-3">
+      <label *ngIf="props.label" [for]="field.id">
+        {{ props.label + (props.required ? " *" : "") }}
+      </label>
+
+      <div class="form-control input-group p-0">
+        <select
+          class="form-select"
+          [formControl]="asFormControl(formControl)"
+          [formlyAttributes]="field"
+          (ngModelChange)="updateSelectedLicense($event)"
+        >
+          @for (license of availableLicenses() | keyvalue; track license) {
+          <option [value]="license.key">
+            {{ license.value.name }}
+          </option>
+          }
+        </select>
```

Comment:
would it be easier to use a typeahead here?

Reply:
I did consider it. The reason why I chose against using a typeahead is because it requires you to know what license you are looking for.

And regular select boxes also support typing to find options

Reply:
I also thought about this: it should be less needed with just the data-oriented licenses in the list